src/tasks/LibraryBuilder/Templates/preloaded-assemblies.c

```diff
@@ -9,7 +9,21 @@
 static void
 preload_assembly (const char* filename)
 {
-    MonoAssembly *assembly = mono_assembly_load_with_partial_name (filename, NULL);
+    MonoAssembly *assembly = mono_assembly_open (filename, NULL);
+    if (assembly)
+        return;
+
+    int len = strlen (filename);
+    char *filename_without_extension = strdup (filename);
+    if (!filename_without_extension)
+        LOG_ERROR ("Out of memory.\n");
+
+    if (!strcasecmp (".dll", &filename [len - 4]))
```

Comment:
Added a macro to the library-builder for OS specific native functions, but as LibraryBuilder hasn't been extended to support Windows yet, it will always hit the else case for now.